src/app/space/analyze/analyze-overview/analyze-overview.spec.ts

```diff
@@ -0,0 +1,228 @@
+import { AnalyzeOverviewComponent } from './analyze-overview.component';
+
+import {
+  Component,
+  DebugElement,
+  Input
+} from '@angular/core';
+import { TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { Broadcaster } from 'ngx-base';
+import { BsModalService } from 'ngx-bootstrap/modal';
+import { Context, Contexts, Space } from 'ngx-fabric8-wit';
+import { AuthenticationService, User, UserService } from 'ngx-login-client';
+import { ConnectableObservable, Subscription } from 'rxjs';
+import { Observable, Subject } from 'rxjs';
+import { createMock } from 'testing/mock';
+import {
+  initContext,
+  TestContext
+} from 'testing/test-context';
+
+@Component({
+  template: '<alm-analyzeOverview></alm-analyzeOverview>'
+})
+class HostComponent { }
+
+@Component({
+  selector: 'fabric8-edit-space-description-widget',
+  template: ''
+})
+class FakeFabric8EditSpaceDescriptionWidget {
+}
+
+@Component({
+  selector: 'fabric8-work-item-widget',
+  template: ''
+})
+class FakeFabric8WorkItemWidget {
+}
+
+@Component({
+  selector: 'fabric8-pipelines-widget',
+  template: ''
+})
+class FakeFabric8PipelinesWidget {
+}
+
+@Component({
+  selector: 'fabric8-add-codebase-widget',
+  template: ''
+})
+class FakeFabric8AddCodebaseWidget {
+}
+
+@Component({
+  selector: 'fabric8-analytical-report-widget',
+  template: ''
+})
+class FakeFabric8AnalyticalReportWidget {
+}
+
+@Component({
+  selector: 'fabric8-create-work-item-widget',
+  template: ''
+})
+class FakeFabric8CreateWorkItemWidget {
+}
+
+@Component({
+  selector: 'fabric8-environment-widget',
+  template: ''
+})
+class FakeFabric8EnvironmentWidget {
+}
+
+@Component({
+  selector: 'f8-feature-toggle',
+  template: ''
+})
+class FakeFabric8FeatureToggle {
+}
+
+@Component({
+  selector: 'flow-selector',
+  template: ''
+})
+class FlowSelector {
+  @Input() space: Space;
+}
+
+@Component({
+  selector: 'import-wizard',
+  template: ''
+})
+class ImportWizard {
+}
+
+@Component({
+  selector: 'quickstart-wizard',
+  template: ''
+})
+class QuickstartWizard {
+}
+
+describe('AnalyzeOverviewComponent', () => {
+  type ctx = TestContext<AnalyzeOverviewComponent, HostComponent>;
+
+  let modalService: jasmine.SpyObj<BsModalService>;
+  let broadcaster: jasmine.SpyObj<Broadcaster>;
+  let authentication: jasmine.SpyObj<AuthenticationService>;
+  let ctxSubj: Subject<Context>;
+
+  let authenticationServiceMock = (): jasmine.SpyObj<AuthenticationService> => {
+    authentication.isLoggedIn = function() { return true; } as (() => boolean) & jasmine.Spy;
+    return authentication;
+  };
```

Comment:
This is a little awkward. This is named as a mock but is not actually a mock - it's a function that returns a mock. Why does this need to be extracted out rather than inlined into the module configuration on line 160?